app/models/product.rb

```diff
@@ -127,7 +127,17 @@ def self.recommended_extensions(root_product_ids)
   end
 
   def service
-    Service.find_or_create_by(product_id: id)
+    service = Service.find_by(product_id: id)
+
+    service ||= if Service.find_by(id: id)
+                  # for backward-compatibility: create a service with autoincrement ID, if product.id is already occupied
+                  Service.create!(product_id: id)
+                else
+                  # try to create a service with service.id = product.id for keeping consistent service URLs across RMT instances
+                  Service.create!(id: id, product_id: id)
```

Comment:
There is the chance of a race condition here. Could it happen that on a system this method gets called in parallel?

Reply:
Indeed, there is a chance of that. I'll move this into `RepositoryService` so that it's called from `rmt-cli sync` only and wrap the whole product creation method into a transaction.